source/llassetgen-cmd/source/main.cpp

```diff
@@ -0,0 +1,226 @@
+#if _MSC_VER >= 1900  // handle Visual Studio bug
+    std::wstring_convert<std::codecvt_utf8<uint32_t>, uint32_t> ucs4conv;
+    auto convStr = ucs4conv.from_bytes(str);
+    return std::u32string(reinterpret_cast<const char32_t*>(convStr.data()), convStr.size());
+#else
+    std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> ucs4conv;
+    return ucs4conv.from_bytes(str);
+#endif
+}
+
+std::vector<Vec2<size_t>> sizes(const std::vector<Image>& images, unsigned int downsamplingRatio) {
+    std::vector<Vec2<size_t>> imageSizes(images.size());
+    std::transform(images.begin(), images.end(), imageSizes.begin(),
+                   [downsamplingRatio](const Image& img) { return img.getSize() / downsamplingRatio; });
+    return imageSizes;
+}
+
+std::pair<std::string, std::string> outNames(const std::string& outPath) {
+    std::string pathWithoutExtension;
+    if (outPath.substr(outPath.length() - 4) == ".png") {
+        pathWithoutExtension = outPath.substr(0, outPath.length() - 4);
+    } else {
+        pathWithoutExtension = outPath;
+    }
+    return std::make_pair(pathWithoutExtension + ".png", pathWithoutExtension + ".fnt");
+}
+
+std::set<unsigned long> makeGlyphSet(const std::string& glyphs, const std::vector<unsigned int>& charCodes,
+                                     bool includeAscii) {
+    std::set<unsigned long> set;
+    for (const auto c : UTF8toUCS4(glyphs)) {
+        set.insert(c);
+    }
+
+    for (const auto c : charCodes) {
+        set.insert(c);
+    }
+
+    if (includeAscii) {
+        const char* p = ascii;
+        while (*p) {
+            set.insert(static_cast<unsigned long>(*p++));
+        }
+    }
+    return set;
+}
+
+int parseAtlas(int argc, char** argv) {
+    // Example: llassetgen-cmd atlas -d parabola --ascii -f Verdana atlas.png
+    CLI::App app{atlasHelp};
+
+    // positional arguments
+    std::string outPath;
+    app.add_option("outfile", outPath, aOutfileHelp)->required();
+
+    // algorithms
+    std::string algorithm;
+    CLI::Option* distfieldOpt = app.add_set("-d, --distfield", algorithm, algoNames(dtAlgos), distfieldHelp);
+
+    std::string packing = "shelf";
+    app.add_set("-k, --packing", packing, algoNames(packingAlgos), packingHelp, true);
+
+    // glyphs
+    std::string glyphs;
+    CLI::Option* glyphsOpt = app.add_option("-g, --glyph", glyphs, glyphHelp);
+
+    std::vector<unsigned int> charCodes;
+    app.add_option("-c, --charcode", charCodes, charcodeHelp);
+
+    bool includeAscii = false;
+    app.add_flag("--ascii", includeAscii, asciiHelp);
+
+    // font
+    unsigned int fontSize = 128;
+    app.add_option("-s, --fontsize", fontSize, fontsizeHelp, true);
+
+    std::string fontName;
+    CLI::Option* fontNameOpt = app.add_option("-f, --fontname", fontName, fontnameHelp);
+
+    std::string fontPath;
+    CLI::Option* fontPathOpt = app.add_option("--fontpath", fontPath, fontpathHelp)->check(CLI::ExistingFile);
+
+    // TODO: app.requires_one
```

Comment:
How can this TODO be resolved? Or what you are waiting for to resolve this? (add the answer as comment, so that later developers might understand what you meant with `app.requieres_one`.

Reply:
Right, I'd forgotten about that. I'm gonna resolve it now, it's quite a severe bug.